packages/core/src/types/provider.ts

```diff
@@ -302,6 +302,29 @@ export type Asset = {
       url?: string;
       /** URL to access file via HTTP through an IPFS gateway */
       gatewayUrl?: string;
+      spec?: {
+        /**
+         * Name of the NFT metadata template to export. 'player' will embed the
+         * Livepeer Player on the NFT while 'file' will reference only the
+         * immutable MP4 files.
+         */
+        nftMetadataTemplate?: 'player' | 'file';
+        /**
+         * Additional data to add to the NFT metadata exported to IPFS. Will be
+         * deep merged with the default metadata exported.
+         */
+        nftMetadata?: {
+          [k: string]: unknown;
+        };
+      };
```

Comment:
So do you think it's OK to go with this current API? Any things [1] you'd prefer to change?

Anyway, if we keep it as is, one thing that we also need to change is the `AssetPatchPayload`. Check out the one from Studio, we should give the option for callers to set this `spec` field in case they want to configure the generated NFT metadata.

[1] Considering we don't have a lot of extra bandwidth though. I could probably only pick up some small tweaks here or there, or help you make the changes yourself on the API.

Reply:
Yeah that makes sense! We should update the types to support nft metadata. Given that bandwidth is limited and we have a few other ongoing tasks before devcon, does it make sense to add an issue in the backlog and follow up later? And get these simple docs in

Reply:
I think there are 2 follow-ups here:
 - Adding support on the SDK for customizing the NFT metadata. This should be done immediately, on this PR, by updating the `AssetPatchPayload` types to support customizing the metadata (apart from only setting it to `true` and using default values). I believe this is an important aspect of the experience here so apps are not restricted to the format we made for the metadata. We even say in the docs that "the metadata can also be customized to override any of these default fields", so we ned to allow that on the patch input as well (or fix the docs, but as I said I think it's important for it to be customizable).
 - Upgrading this API, likely in a backward incompatible way, to something more ergonomic. I think this is the part that we should leave for later, maybe open an issue for it (tho I think we may already have it). This is lower pri given the current bandwidth etc.

Something we _could_ do that is relatively small is create an `empty` value for `nftMetadataTemplate` as well, in which you could set the entire `nftMetadata` from scratch and don't get it changed by the API in any way. There would be a problem in how would you inject the CID of the exported video in that metadata, but there can be some workarounds (worst case just make the request twice, once to get cid another to create metadata with that cid). Do you think something like that would be necessary? Would avoid user having to override every field generated by default on the API (setting it to `null` if they want to not have it)

Reply:
Sure, I can add that - I agree the metadata will need rethinking and I don't think this PR is the place for that. Updating the types now.

Reply:
Done